### ISSUE TITLE
Question on implementing predcls in OpenImages

### ISSUE THREAD

**Issue by WonwoongCho** [hesedjds/SQUAT#2]
Hi, thank you for sharing the awesome project.
I was wondering if I could get some pieces of information to implement predcls task in OpenImages dataset.
Here are several questions I have. Please take a quick pass:

1. I roughly guess I might need to replace the Node Detection results with embeddings from GT boxes. Does it make sense? Would there be other things that need to be changed, from your perspective? 
2. If I made some changes to the code like mentioned above, can I still use the same command in scripts/test.sh?
3. Can I use the pretrained model (in this repo, trained with openImages) when implementing predcls task? Or do I need to train another model specifically for predcls task?
4. Can I use the ipynb file for visualization (https://github.com/KaihuaTang/Scene-Graph-Benchmark.pytorch/tree/master/visualization)? Or do you have your own visualization code?

Maybe some of the questions are not easy to answer. I would appreciate it if you could answer some of the questions if it does not take that much time. Thanks!

**Comment by hesedjds** [comment]
I appreciate your interest. 
1. Yes, it makes sense. The SGCls and PredCls settings already use GT boxes. 
You can easily replace the Node Detection results with embeddings from GT boxes by config MODEL.ROI_RELATION_HEAD.USE_GT_BOX (If you want to use GT labels, please check MODEL.ROI_RELATION_HEAD.USE_GT_LABEL). 

2. If you use the config mentioned, you can use the same command in scripts/test.sh. 
I cannot say the code works if you made the other changes.

3. Maybe possible. 
I observed that the SGDet-pre-trained model can cover the simpler task (SGCls and PredCls) to some extent. 
However, if you want higher performance, you should specifically train another model for the PredCls task. 

4. Scene-Graph-Benchmark.pytorch has the same structure as my code base (PySGG); thus, maybe you can use the ipynb file for visualization in Scene-Graph-Benchmark.pytorch. However, I am unsure since I did not try to use them. 
In the paper, we check the result of top-K triplets and draw the figure manually.

**Comment by hesedjds** [comment]
For the OpenImages v6, MODEL.ROI_RELATION_HEAD.USE_GT_BOX and MODEL.ROI_RELATION_HEAD.USE_GT_LABEL also work. 
Thus, you can use the config mentioned above if you want to train the model, pre-trained on OpenImages v6, on SGCls, or PredCls setting. 
However, the evaluation code for OpenImages v6 of the codebase (PySGG) is not for the SGCls and PredCls settings. 
Since I have not tried it before, I am not sure whether the evaluation code works on the SGCls and PredCls settings or not.

**Comment by WonwoongCho** [comment]
Alright, thank you for sharing your insight!!
